test/junit/scala/tools/nsc/backend/jvm/BytecodeTest.scala

```diff
@@ -529,6 +529,41 @@ class BytecodeTest extends BytecodeTesting {
     // ---------------
 
     assertSameCode(getMethod(fooClass, "m1"), List(
+VarOp(ALOAD, 1),
+VarOp(ASTORE, 3),
+VarOp(ALOAD, 3),
+TypeOp(INSTANCEOF, "scala/collection/immutable/$colon$colon"),
+Jump(IFEQ, Label(19)),
+VarOp(ALOAD, 3),
+TypeOp(CHECKCAST, "scala/collection/immutable/$colon$colon"),
+VarOp(ASTORE, 4),
+VarOp(ALOAD, 4),
+Invoke(INVOKEVIRTUAL, "scala/collection/immutable/$colon$colon", "head", "()Ljava/lang/Object;", false),
+Invoke(INVOKESTATIC, "scala/runtime/BoxesRunTime", "unboxToInt", "(Ljava/lang/Object;)I", false),
+VarOp(ISTORE, 5),
+VarOp(ILOAD, 5),
+Op(IRETURN),
+Label(19),
+Jump(GOTO, Label(22)),
+Label(22),
+Field(GETSTATIC, "scala/collection/immutable/Nil$", "MODULE$", "Lscala/collection/immutable/Nil$;"),
+VarOp(ALOAD, 3),
+Invoke(INVOKEVIRTUAL, "java/lang/Object", "equals", "(Ljava/lang/Object;)Z", false),
+Jump(IFEQ, Label(31)),
+IntOp(BIPUSH, 20),
+Op(IRETURN),
+Label(31),
+Jump(GOTO, Label(34)),
+Label(34),
+TypeOp(NEW, "scala/MatchError"),
+Op(DUP),
+VarOp(ALOAD, 3),
+Invoke(INVOKESPECIAL, "scala/MatchError", "<init>", "(Ljava/lang/Object;)V", false),
+Op(ATHROW),
+
+
+
```

Comment:
To be reformatted. The test diff output makes updating the test difficult. Oh, I guess it's called diff for difficult.